sycl/include/sycl/usm.hpp

```diff
@@ -19,280 +20,248 @@ namespace sycl {
 ///
 // Explicit USM
 ///
-__SYCL_EXPORT void *malloc_device(
-    size_t size, const device &dev, const context &ctxt,
-    const detail::code_location CodeLoc = detail::code_location::current());
-__SYCL_EXPORT void *malloc_device(
-    size_t size, const device &dev, const context &ctxt,
-    const property_list &propList,
-    const detail::code_location CodeLoc = detail::code_location::current());
-__SYCL_EXPORT void *malloc_device(
-    size_t size, const queue &q,
-    const detail::code_location CodeLoc = detail::code_location::current());
-__SYCL_EXPORT void *malloc_device(
-    size_t size, const queue &q, const property_list &propList,
-    const detail::code_location CodeLoc = detail::code_location::current());
+__SYCL_EXPORT void *malloc_device(size_t size, const device &dev,
+                                  const context &ctxt _CODELOCPARAM(&CL));
+__SYCL_EXPORT void *
+malloc_device(size_t size, const device &dev, const context &ctxt,
+              const property_list &propList _CODELOCPARAM(&CL));
+__SYCL_EXPORT void *malloc_device(size_t size,
+                                  const queue &q _CODELOCPARAM(&CL));
+__SYCL_EXPORT void *
+malloc_device(size_t size, const queue &q,
+              const property_list &propList _CODELOCPARAM(&CL));
 
-__SYCL_EXPORT void *aligned_alloc_device(
-    size_t alignment, size_t size, const device &dev, const context &ctxt,
-    const detail::code_location CodeLoc = detail::code_location::current());
-__SYCL_EXPORT void *aligned_alloc_device(
-    size_t alignment, size_t size, const device &dev, const context &ctxt,
-    const property_list &propList,
-    const detail::code_location CodeLoc = detail::code_location::current());
-__SYCL_EXPORT void *aligned_alloc_device(
-    size_t alignment, size_t size, const queue &q,
-    const detail::code_location CodeLoc = detail::code_location::current());
-__SYCL_EXPORT void *aligned_alloc_device(
-    size_t alignment, size_t size, const queue &q,
-    const property_list &propList,
-    const detail::code_location CodeLoc = detail::code_location::current());
+__SYCL_EXPORT void *
+aligned_alloc_device(size_t alignment, size_t size, const device &dev,
+                     const context &ctxt _CODELOCPARAM(&CL));
```

Comment:
nit: I don't think reference `&` makes any change, neither for `&CL` nor for `&propList` below. It's also harmless though.